astropy/coordinates/tests/test_iau_fullstack.py

```diff
@@ -166,7 +165,7 @@ def test_future_altaz():
     if hasattr(utils, '__warningregistry__'):
         utils.__warningregistry__.clear()
 
-    with catch_warnings() as found_warnings:
+    with pytest.warns(AstropyWarning) as found_warnings:
```

Comment:
I'd write this with an `ExitStack` context (with entries added as per the `if-then` logic below), but again no problem to just keep it.

EDIT: ie., basically like you do below!

Reply:
Nesting doesn't seem to work as expected. The second context can't see the warnings.

Reply:
O, sorry to hear that (and sorry for wasting your time...)

Reply:
No worries! It was a great idea and would have been very elegant if it were to work.